### PR TITLE
Use lumino

### LINKED ISSUE #7534: Switch to lumino
We have continued the Phosphor project code base as a new project, [lumino](https://github.com/jupyterlab/lumino).  We have released all of the `lumino` packages at the same version as they were previously in Phosphor.  The plan is for 2.0 to depend on `lumino`, and resolve any `@phosphor` libraries to `lumino` using `yarn` resolutions.

- [ ] Migrate code to point to `@lumino` packages
- [ ] Add yarn resolution for `@phosphor` -> `@lumino`
- [ ] Update documentation

### PR DESCRIPTION
**Pull request by blink1073**
## References
Fixes #7534 

## Code changes
All imports now use `@lumino` packages.  Extension authors will have to switch from `@phosphor` to `@lumino`.

## User-facing changes
None.

## Backwards-incompatible changes
None, extensions can continue to use `@phosphor` packages for one more major release.

### CODE REVIEW

File: buildutils/src/dependency-graph.ts [hunk 1]
```diff
@@ -277,7 +277,7 @@ commander
     '--no-jupyterlab',
     'Do not include dependency connections TO @jupyterlab org packages nor isolated @jupyterlab org packages'
   )
-  .option('--no-phosphor', 'Do not include @phosphor org packages')
+  .option('--no-phosphor', 'Do not include @lumino org packages')
```
Comment (afshin) [inline buildutils/src/dependency-graph.ts]: Should we have two options here? `--no-phosphor` as a legacy and `--no-lumino` as the new option?

File: dev_mode/webpack.config.js [hunk 2]
```diff
@@ -154,6 +154,67 @@ module.exports = [
     entry: {
       main: ['whatwg-fetch', plib.resolve(buildDir, 'index.out.js')]
     },
+    // Map Phosphor files to lumino files.
```
Comment (afshin) [inline dev_mode/webpack.config.js]: `lumino` => `Lumino`

**Comment by timkpaine** [comment]
This requires a major version bump (e.g. 2.0 and 1.3) as its not backwards compatible, right?

**Comment by blink1073** [comment]
Good catch, thanks @timkpaine - we'll have to bump all of the JLab packages.  The production webpack settings are derived from the development ones, so the aliases will also work.

**Comment by blink1073** [comment]
It looks like @jasongrout already bumped all of our package by a major version on 1 Oct in https://github.com/jupyterlab/jupyterlab/commit/97e3ff073b4c51f0f816f739b338c32b7a03c44e#diff-73791fc5c47cb5da6dfba8e7d88466fc.

**Comment by jasongrout** [comment]
We were hoping to ease porting of extensions by automatically mapping phosphor packages to lumino packages. However, if this is tricky for us to do, I'm perfectly fine with telling people to replace their phosphor packages with lumino for jlab 2.0, and not providing this shim for phosphor dependencies.

**Comment by afshin** [comment]
> We were hoping to ease porting of extensions by automatically mapping phosphor packages to lumino packages. However, if this is tricky for us to do, I'm perfectly fine with telling people to replace their phosphor packages with lumino for jlab 2.0, and not providing this shim for phosphor dependencies.

I think it would probably be less pain in the long run to ask extension authors to replace `@phosphor/` with `@lumino/`.

**Comment by jasongrout** [comment]
> I think it would probably be less pain in the long run to ask extension authors to replace `@phosphor/` with `@lumino/`.

And less magic, which means less confusion and less time from us in answering questions from confused people.

**Pull request merged**